internal/errors/option.go

```diff
@@ -46,3 +52,22 @@ func WithOp(op Op) Option {
 		o.withOp = op
 	}
 }
+
+// WithCallerOp sets the `withOp` field to the caller's functions name; formatted package.func.
+// Optionally, with fullpath=true, it will include the full package path.
+// If caller lookup fails this will return an empty string
+func WithCallerOp(fullpath bool) Option {
```

Comment:
Helper function so we can have CallersFrames do the lifting for us here

Reply:
Some idea of the output of this with these changes and forced errors:
fullpath=true
```
Error from controller when performing create on static-type host: 
Error information:
  Code:                InvalidArgument
  Message:             github.com/hashicorp/boundary/internal/host/static.(*Repository).CreateHost:(54): create: static host: parameter violation: error #100
  Status:              400
```

fullpath=false
```
Error from controller when performing create on static-type host: 
Error information:
  Code:                InvalidArgument
  Message:             static.(*Repository).CreateHost (54): create: static host: parameter violation: error #100
  Status:              400
```